File: python/data_wilayah_indonesia/src/data_wilayah_indonesia/core/entities/regency.py
from dataclasses import dataclass
from enum import Enum
from .administrative_unit import AdministrativeUnit

class RegencyType(Enum):
    KABUPATEN = "KABUPATEN"
    KOTA = "KOTA"

@dataclass(frozen=True)
class Regency(AdministrativeUnit):
    """
    Entitas untuk tingkat Kabupaten/Kota.
    """
    province_code: str
    type: RegencyType

    def __post_init__(self):
        super().__post_init__()
        if not self.province_code:
            raise ValueError("Province code tidak boleh kosong.")
        if not isinstance(self.type, RegencyType):
            raise ValueError("Regency type tidak valid.")

File: packages/data-wilayah-indonesia/src/core/entities/regency.ts
import { AdministrativeUnit, RegencyType } from "./administrative-unit";

/**
 * Regency/City entity.
 */
export class Regency extends AdministrativeUnit {
  constructor(
    code: string,
    name: string,
    public readonly provinceCode: string,
    public readonly type: RegencyType
  ) {
    super(code, name);
    if (!provinceCode) throw new Error("Province code cannot be empty.");
  }
}
